src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -226,6 +226,29 @@ public ConfidentialClientApplicationBuilder WithClientAssertion(Func<string> cli
             return this;
         }
 
+        /// <summary>
+        /// Configures a delegate that creates a client assertion. See https://aka.ms/msal-net-client-assertion
+        /// </summary>
+        /// <param name="clientAssertionDelegate">delegate computing the client assertion used to prove the identity of the application to Azure AD.
+        /// This is a delegate that computes a Base-64 encoded JWT for each authentication call.</param>
+        /// <returns>The ConfidentialClientApplicationBuilder to chain more .With methods</returns>
+        /// <remarks> Callers can use this mechanism to cache their assertions </remarks>
+        public ConfidentialClientApplicationBuilder WithClientAssertion(Func<AssertionRequestOptions, string> clientAssertionDelegate)

```

Comment:
I'd probably refactor the existing `WithClientAssertion` overload to use this mechansm now. And probably hide the old mechanism (via EditorBrowsable=false).

Reply:
The idea was to not break existing functionality. Marking it as EditorBrowsable=false is a great idea. Had not thought about that.

How do the documentation pages get updated? Because if we mark this as not browseable the docs won't match the functions.